Source/gg2/Objects/Weapons/Medigun.events/User Event 1.xml

```diff
@@ -16,12 +16,32 @@
         <argument kind="STRING">{
     if(!instance_exists(healTarget) and readyToShoot)
     {
-        var bestTarget, bestTargetDist;
+        var bestTarget, bestTargetDist, testx, testy;
         bestTarget=noone;
-        bestTargetDist=maxHealDistance;
+        bestTargetDist = maxHealDistance;
+        
+        testx = round(owner.x) + cos(degtorad(owner.aimDirection))*owner.aimDistance;
+        testy = round(owner.y) - sin(degtorad(owner.aimDirection))*owner.aimDistance;
 
         with(Character) {
-            if(id != other.owner and team == other.owner.team) {
+            if point_distance(x,y,other.owner.x,other.owner.y) &lt;= other.maxHealDistance
+               &amp;&amp; point_distance(x,y,testx,testy) &lt; bestTargetDist {
+                if(id != other.owner and (team == other.owner.team)) {
+                    if(collision_line(other.x, other.y,
+                    other.x+lengthdir_x(other.maxHealDistance, other.owner.aimDirection),
+                    other.y+lengthdir_y(other.maxHealDistance, other.owner.aimDirection), id, true, false))
+                    {
+                        if(!collision_line(other.x, other.y, x, y, Obstacle, true, true)) {
+                            bestTargetDist = point_distance(x,y,testx,testy);
+                            bestTarget = player;
+                        }
+                    }
+                }
+            }
+        }
+        
+        if bestTarget = noone with(Character) {
```

Comment:
I went through the logic and most of the conditions for the second attempt of finding a heal target are identical. The only way the second attempt can find a target when the first run didn't would be if the cursor was more than `maxHealDistance` from the nearest target - ok, but seems somewhat arbitrary to use that particular distance for the cursor cut-off. I'd actually expect something a bit smaller.
IMO it would be ok to leave the two attempts, but the code should be deduplicated. We could e.g. build a list of eligible characters first and then run through that list twice. Alternatively, run through the characters once and filter the eligible characters in some outer if-statement(s), then calculate the results for both "closest" metrics in parallel inside that and decide which to actually use after the loop is done.